feat: add slack integration sections (#590)
tolgee/documentation: platform/integrations/slack_integration/setup.mdx

---

> id: setup

Reviewer: Can we plesae also small "About" page as a home for the slack integration covering the capabilities of the integration. It's goog for SEO, but also it's good when we want to link to it from somewhere, so we don't have to directly point to setup.

title: Setup Tolgee Slack Integration
sidebar_label: Setup
description: Learn how to install and setup Tolgee Slack app.
---

import { ScreenshotWrapper } from '../../shared/_ScreenshotWrapper';

## Prerequisites

Before you begin, ensure you have:

- A Tolgee account with the necessary permissions to manage integrations
- A Slack workspace where you have permission to add apps

:::info Feature availability

Tolgee Slack Integration is available in the Business and Enterprise plans. [Upgrade your plan](/pricing) to use this feature.

If you use the self-hosted version, you need to [set up the license](/platform/self_hosting/licensing) to use this feature.

:::

> ## Setup for Cloud Users

Reviewer: Cloud is a priority for us, so I would remove this headline. And move the steps one level higher.
The Self-hosted setup might be moved to the self-hosting section. Here we can only add link to it.

### Step 1: Install the Tolgee App in Slack

First of all you need to install the Tolgee app in your Slack workspace. You can do that right via our web app:

> 1. Open your [Tolgee Project](https://app.tolgee.io/projects) (login if you are not already)

Reviewer: ```suggestion
1. Open [Tolgee Platform](https://app.tolgee.io/) (login if you are not already)
```

> 2. Click on your icon in the right top corner.

Reviewer: ```suggestion
2. Click on your icon in the right top corner and go to **Organization Settings**
```

3. Then navigate to **Apps**.
4. Now you will see all available apps. Click on **Connect To Slack** button in the **Slack** app.

<ScreenshotWrapper src="/img/docs/platform/slack_integration/install_app.png" />

After that follow the prompts to authorize Tolgee app to access your Slack workspace just by clicking **Allow**

<ScreenshotWrapper src="/img/docs/platform/slack_integration/authorize.png" />

:::info Info

The first step must be done by the **organization owner**. The second step must be done by any user to access the commands.

:::

> ### Step 2: Connect Slack workspace with your Tolgee account

Reviewer: ```suggestion
### Step 2: Connect your Slack account with your Tolgee account
```

> After installing the Tolgee app, you can interact with it either as a personal app or within a channel. Once the app is installed in the workspace, it will be available in all public channels.

Reviewer: This paragraph should be moved to the upper section.

For private channels, you need to invite the Tolgee app manually by using the command `/invite @Tolgee`.

> Now you need to link your Tolgee account with the Slack workspace:

Reviewer: ```suggestion
Now you need to link your Tolgee account with your Slack account:
```

Reviewer: It would be also nice to describe why.

1. Type `/tolgee login` in the chat room to which the Tolgee app is invited.
2. Button **Connect Tolgee account** will appear in the chat.

<ScreenshotWrapper src="/img/docs/platform/slack_integration/slack_login.png" />

3. Click on it and you will be redirected to the Tolgee login page.

<ScreenshotWrapper src="/img/docs/platform/slack_integration/tolgee_login.png" />

4. Then click **Link Accounts**.

After login you will see a success message in the chat.

Now you can use the Tolgee app in your Slack workspace.

> ## Setup For Self-Hosted Users

Reviewer: Please move this to the self-hosting section or other page, as discussed before.

If you are using the self-hosted version of Tolgee, you also can use the Slack integration. The setup is a bit different than for cloud users.

### Step 1: Create and Install the Tolgee App in Slack

> First of all you need to create a new app in your Slack workspace. You can do that by navigating to [Slack API](https://api.slack.com/apps)

Reviewer: ```suggestion
First of all you need to create a new app in your Slack workspace. You can do that by navigating to [Slack Apps page](https://api.slack.com/apps)
```

2. Navigate to your Tolgee instance and open `application.yaml` file.
3. Add the following configuration:
```yaml
tolgee:
 slack:
    signing-secret: <signing-secret>
```

Your next step depends on whether you are using the Slack with single workspace or with multiple workspaces.
In most cases self-hosted instances don't need to connect to multiple slack workspaces.

#### Single Workspace

1. First you need to provide all necessary bot Scopes. Navigate to **OAuth & Permissions** page and add the following scopes:
   - `channels:history`
   - `channels:read`
   - `chat:write`
   - `chat:write.customize`
   - `commands`
   - `groups:history`
   - `groups:read`
   - `im:history`
   - `im:read`
   - `mpim:history`
   - `mpim:read`
   - `users:read`

2. Then reinstall the app in your workspace by clicking on **Reinstall App** button. This is necessary to apply the new scopes.
Now you can copy the **Bot User OAuth Token**. It starts with `xoxb-`.

3. Now you need to provide the token to your Tolgee instance. Navigate to your Tolgee instance and open `application.yaml` file.
Add the following configuration:
```yaml
tolgee:
 slack:

>     bot-token: <your-bot-token>

Reviewer: As stated below, we can remove the multi workspace setup. So we can merge the configuration together, so user can configure it in one step.

```

> #### Multiple Workspace

Reviewer: We can remove this section, users probably won't need it and it complicates the docs.

---
id: setup

> title: Setup

Reviewer: This should be `Setup Tolgee Slack Integration`, for better SEO

sidebar_label: Setup

> description: Learn how to install and setup Tolgee Slack app. First, you need to install the app from the Slack App Directory by clicking on the Add to Slack button.

Reviewer: This should be short description. Remove there is the `Frst...` part.

---

import { FigmaScreenshotWrapper } from '../../shared/_FigmaScreenshotWrapper';

> 

Reviewer: We should tell the user that slack is available in advanced subscription plans and link to the pricing. Like here: https://tolgee.io/platform/projects_and_organizations/members#granular-permissions

> ## Installing the Tolgee Integration for Slack

Reviewer: This subtitle seems unnecesary to me.

> 

Reviewer: Somewhere should be covered the Slack configuration for self-hosting.

### Prerequisites

Before you begin, ensure you have:

> - A Tolgee account with the necessary permissions to manage integrations.

Reviewer: Points like these are not sentences, so they doesn't end with `.`

- A Slack workspace where you have permission to add apps.

### Step-by-Step Guide

#### Step 1: Install the Tolgee App in Slack

> Install Slack application from [Slack App Directory](https://slack.com/apps/dumbLink) by clicking on the **Add to Slack** button. Or you can find app right in Slack by

Reviewer: This is not the flow we want the user to follow. We want to install the app from Tolgee platform and it will install it to the workspace. This should contain a screenshot.

opening the Apps > Add apps at your workspace and searching for "**Tolgee**", and then click **Add**.

After that follow the prompts to authorize Tolgee app to access your Slack workspace just by clicking **Allow**

> <FigmaScreenshotWrapper src="/img/docs/platform/slack_integration/authorize.png" />

Reviewer: The `FigmaScreenshotWrapper` is for Figma integration screenshots.
